server/controllers/user.ts

```diff
@@ -59,6 +58,14 @@ export function displayRegisterPage(req:Request, res: Response, next: NextFuncti
 
 }
 
+export function displayUserEditPage(req:Request, res: Response, next: NextFunction): void {
+    if(!req.user)
```

Comment:
This part is not needed.

1. We should already have the authGuard middleware in our routes file. It should be guaranteed that the user is logged in at this point.
1. We also need to add a check on whether the requesting user is the same one with the requested id.
1. This route handler is not being used yet. Check the `server/routes/user.ts` file.
1. `!req.user` is the same as `!req.isAuthenticated()`, which returns true when the user is not logged in. This will result in not returning a response to the user request if the user is logged in.